src/Signer/Ecdsa.php

```diff
@@ -29,6 +32,8 @@ final public function sign(string $payload, Key $key): string
 
     final public function verify(string $expected, string $payload, Key $key): bool
     {
+        assert($expected !== '');
```

Comment:
Wait, is the signature not supposed to be non-empty upfront?

Reply:
Not until we allow `\Lcobucci\JWT\Signer\None` to exist.

`\Lcobucci\JWT\Signer::verify` API must take into account all available signers.